### ISSUE TITLE
Autocorrect for `Style/TrailingBodyOnClass` breaks the code when body trails after singleton class definition

### ISSUE THREAD

**Issue by johnny-miyake** [rubocop/rubocop#10618]
Related with https://github.com/rubocop/rubocop/pull/10617

The autocorrect for [Style/TrailingBodyOnClass](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/TrailingBodyOnClass) breaks this code
```ruby
class Foo bar # <------------ `bar` is the actual first body line of this class.
  class << self; baz  # <------------ `baz` is the actual first body line of this singleton class.
    def func
      puts 1
    end
  end
end
```
to
```ruby
class Foo 
  bar
  class << self baz # Syntax error
    def func
      puts 1
    end
  end
end
```

## Expected behavior

```ruby
class Foo 
  bar    
  class << self
    baz
    def func
      puts 1
    end
  end
end
```

## Steps to reproduce the problem
Save the example code to example.rb, and run
```sh
exe/rubocop example.rb -a --only Style/TrailingBodyOnClass 
```

## RuboCop version

```sh
$ bundle exec rubocop -V
1.29.0 (using Parser 3.1.2.0, rubocop-ast 1.17.0, running on ruby 2.7.5 x86_64-darwin20)
  - rubocop-performance 1.13.3
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.10.0
```

**Comment by johnny-miyake** [comment]
There is another pattern.

This code
```ruby
class Bar def bar; end
end
```
is corrected to
```ruby
class Bar
  def bar end # Syntax error
end
```
